File: lib/core/utils/sms_parser/engines/text_normalizer.dart
class TextNormalizer {
  static String normalize(String rawSms) {
    String text = rawSms.toLowerCase();
    
    // Normalize line breaks and tabs to spaces
    text = text.replaceAll(RegExp(r'[\r\n\t]+'), ' ');
    
    // Normalize multiple spaces
    text = text.replaceAll(RegExp(r'\s{2,}'), ' ');
    
    // Normalize currency symbols and words
    text = text.replaceAll('₹', 'rs ');
    text = text.replaceAll('inr', 'rs ');
    
    // Normalize quotes
    text = text.replaceAll(RegExp(r'[`´‘’]'), "'");
    text = text.replaceAll(RegExp(r'[“”]'), '"');
    
    return text.trim();
  }
}

File: android/app/src/main/kotlin/com/smart_money_tracker/parser/TextNormalizer.kt
package com.smart_money_tracker.parser

object TextNormalizer {
    fun normalize(rawSms: String): String {
        var text = rawSms.lowercase()
        
        // Normalize line breaks and tabs to spaces
        text = text.replace(Regex("[\\r\\n\\t]+"), " ")
        
        // Normalize multiple spaces
        text = text.replace(Regex("\\s{2,}"), " ")
        
        // Normalize currency symbols and words
        text = text.replace("₹", "rs ")
        text = text.replace("inr", "rs ")
        
        // Normalize quotes
        text = text.replace(Regex("[`´‘’]"), "'")
        text = text.replace(Regex("[“”]"), "\"")
        
        return text.trim()
    }
}
